### ISSUE TITLE
Date Serialization

### ISSUE THREAD

**Issue by pushplay** [mholt/PapaParse#502]
Hi,

I'm happy with PapaParse but not happy with the date serialization.  The behavior seems to be using Date.toString() which is inconsistent with JSON.stringify(), which uses Date.toISOString().  For example...

```
> let papa = require('papaparse')
undefined
> let data = [{text: "hello world", date: new Date()}]
undefined
> papa.unparse(data)
'text,date\r\nhello world,Fri May 04 2018 14:08:03 GMT-0700 (PDT)'
> JSON.stringify(data)
'[{"text":"hello world","date":"2018-05-04T21:08:03.269Z"}]'
```

I'd like to get consistent behavior with JSON.stringify().  There are a few ways that could happen but the simplest would probably be an `isoDates` boolean option to `unparse`.  If you like the idea I'd be happy to submit a PR.  It looks like a simple change.

**Comment by pokoli** [comment]
I'm wondering if we should not always use JSON.stringify instead of toString(). AFAIU stringify will call toString for most common objects. 

What do you thinks?

P.S: A PR is very welcome, so we can discuss on the code

**Comment by pushplay** [comment]
Doing it across the board without a flag would be a breaking change.  Not a big one but I do want to flag that.

JSON.stringify on a string is going to wrap it in quotes so that will be quite different.

I have a lot of meetings today so we'll see when I can get to a PR.  The more we can agree on in here the less work for me probably.

**Comment by pokoli** [comment]
Don't fear about breaking changes. Probably next version will be a major version (as we have another api changes) so it's a good time to include it.

I do not like the option of using a flag to format date, I will prefer to have only format for dates that work for everyone. So the following questions arise: 

- Why you prefer to use toISOString() instead of toString()?
- Which are the benefits of using toISOStrinng()?
- Is not possible to convert from toString format toISOString()? Is the other way possible?

**Comment by pokoli** [comment]
Before changing the API i will like to know which are the benefits of using ISODate instead of toString

**Comment by pushplay** [comment]
Sure.  There are a few benefits to ISO 8601.  Most importantly it sorts lexicographically so a program can sort on the date column without having to understand the column is a date.  More obviously it's shorter and easier to read.  It's also easier to recognise as a date and parse which makes reversing the operation less painful (see https://github.com/mholt/PapaParse/pull/506/files#diff-eb20bdfe95a768f4eecdf3bebb522a4eR1137).

**Comment by pokoli** [comment]
This has been fixed in #506
